smurfs/src/components/SmurfForm.js

```diff
@@ -0,0 +1,45 @@
+import React        from "react";
+import { connect }  from "react-redux";
+import { addSmurf } from "../actions"
+
+ class SmurfForm extends React.Component {
+    constructor(){
+        super();
+        this.state = {
+            name   : " ",
+            age    : " ",
+            height : " "
+        }
+    }
+    changeHandler = event => {
+        this.setState({
+          ...this.state,
+          [event.target.name]: event.target.value
+        });
+      }
+    submitHandler = event => {
+        event.preventDefault();
+        const { name, age, height } = this.state;
+        this.props.addSmurf({ name, age, height });
```

Comment:
no need o destructure the state object here because you just put it all back together in the addSmurf invocation, you can complete remove line 22 and line 23 would become this.props.addSmurf(this.state)